### PR TITLE
Added missing security checks

### LINKED ISSUE #467: [SecurityBundle] Security: permissions do not have any effect
<a href="https://github.com/turbo-ele"></a>
By @turbo-ele *from* https://github.com/sulu-cmf/SuluSecurityBundle/issues/37

and it should be documented

### PR DESCRIPTION
**Pull request by danrot**
Added the missing security checks for controller and navigation items in admin to Categories, Tags and Snippets.

__Tasks:__

- [x] test coverage
- [ ] gather feedback for my changes

__Informations:__

| Q             | A
| ------------- | ---
| Tests pass?   | yes
| Fixed tickets | fixes #467 
| BC Breaks     | Security have to be set correctly, otherwise users won't be able to work on categories, tags and snippets
| Doc           | none

### CODE REVIEW

File: src/Sulu/Bundle/SnippetBundle/Admin/SuluSnippetAdmin.php [hunk 1]
```diff
@@ -56,4 +69,15 @@ public function getJsBundleName()
     {
         return 'sulusnippet';
     }
+
+    public function getSecurityContexts()
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/SnippetBundle/Admin/SuluSnippetAdmin.php]: provide phpdoc

File: src/Sulu/Bundle/SnippetBundle/Controller/SnippetController.php [hunk 2]
```diff
@@ -426,4 +427,23 @@ private function getReferentialIntegrityResponse($webspace, $references)
...
+     * @param Request $request
+     * @return string
+     */
+    public function getLocale(Request $request)
+    {
+        return $request->query->get('language', null);
+    }
+
+    /**
+     * Returns the SecurityContext required for the controller
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/SnippetBundle/Controller/SnippetController.php]: inheritdoc?

File: src/Sulu/Bundle/TagBundle/Admin/SuluTagAdmin.php [hunk 3]
```diff
@@ -49,4 +62,15 @@ public function getJsBundleName()
     {
         return 'sulutag';
     }
+
+    public function getSecurityContexts()
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/TagBundle/Admin/SuluTagAdmin.php]: phpdoc

File: src/Sulu/Bundle/TagBundle/Controller/TagController.php [hunk 4]
```diff
@@ -317,4 +317,12 @@ public function patchAction(Request $request)
         return $this->handleView($view);
     }
 
+    /**
+     * Returns the SecurityContext required for the controller
+     * @return mixed
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/TagBundle/Controller/TagController.php]: inheritdoc

**Comment by danrot** [comment]
@wachterjohannes Fixed, also for the category bundle ;-)

**Comment by wachterjohannes** [comment]
@danrot perfekt (=

**Pull request merged**